dts/bindings/memory-controllers/atmel,sam-smc.yaml

```diff
@@ -0,0 +1,130 @@
+# Copyright (c) 2022, Basalte bv
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  Atmel Static Memory Controller (SMC).
+
+  The SMC allows to interface with static-memory mapped external devices such as
+  SRAM, PSRAM, PROM, EPROM, EEPROM, LCD Module, NOR Flash and NAND Flash.
+
+  The SMC controller can have up to 4 children defining the connected external
+  memory devices. Device Tree example:
+
+  &smc {
+    status = "okay";
+    pinctrl-0 = <&smc_default>;
+    pinctrl-names = "default";
+
+    is66wv51216dbll@0 {
+      reg = <0>;
+
+      atmel,smc-write-mode = "nwe";
+      atmel,smc-read-mode = "nrd";
+      atmel,smc-setup-timing = <1 1 1 1>;
+      atmel,smc-pulse-timing = <6 6 6 6>;
+      atmel,smc-cycle-timing = <7 7>;
+    };
+  };
+
+  Finally, in order to make the memory available you will need to define new
+  memory device/s in DeviceTree:
+
+  sram1: sram@60000000 {
+      compatible = "zephyr,memory-region", "mmio-sram";
+      device_type = "memory";
+      reg = <0x60000000 DT_SIZE_K(512)>;
+      zephyr,memory-region = "SRAM1";
+  };
+
+compatible: "atmel,sam-smc"
+
+include: [base.yaml, pinctrl-device.yaml]
+
+properties:
+  reg:
+    required: true
+
+  "#address-cells":
+    required: true
+    const: 1
+
+  "#size-cells":
+    required: true
+    const: 0
+
+  peripheral-id:
```

Comment:
Is this actually going into a register somewhere or are you just using it to work around hal limitations? If the former, it's OK, but if the latter, then @galak has a longstanding prohibition against such properties.

Reply:
The `peripheral-id`? This is how all the `sam` SoCs identify a HW block, and isn't something new. I don't want to turn this PR into something bigger.

Reply:
Peripheral-is represents a bit value used to enable clock on the peripheral. For each SoC model this value change the current implementation it is the easy solution to allow share same driver between SoCs and allow DT_INST_FOREACH_STATUS_OKAY for instantiate. I think this property could be removed in the future after implement a clock driver and clock tree.

Reply:
> I think this property could be removed in the future after implement a clock driver and clock tree.

@nandojve based on our offline discussion, it sounds like you're agreeing to take this on so that these properties can be removed treewide for SAM SoCs. Given maintainer commitment to improve this, I am OK with this for now.